graphql_compiler/compiler/emit_sql.py

```diff
@@ -182,6 +182,30 @@ def _find_folded_fields(ir: IrAndMetadata) -> Dict[FoldPath, Set[FoldScopeLocati
     return folded_fields
 
 
+def _get_output_fields_at_fold_location(
+    output_fold_scope_location: FoldScopeLocation,
+    all_folded_fields: Dict[FoldPath, Set[FoldScopeLocation]],
+) -> Set[str]:
+    """Collect output field names for the specified output_fold_scope_location."""
+    # If the location does not have any outputs, return an empty set.
+    if output_fold_scope_location.fold_path not in all_folded_fields:
+        return set()
+    folded_fields = set()
+    for fold_output in all_folded_fields[output_fold_scope_location.fold_path]:
+        # Distinguish folds with the same fold path but different query paths.
+        if (fold_output.base_location, fold_output.fold_path) == (
+            output_fold_scope_location.base_location,
+            output_fold_scope_location.fold_path,
+        ):
```

Comment:
How much simpler would this be if `all_folded_fields` was `Dict[FoldScopeLocation, Set[str]]`? Wouldn't this entire function be just `return all_folded_fields.get(output_fold_scope_location, set())`? I don't understand why we decided `FoldPath` is a good key for `all_folded_fields`.